awswrangler/athena/_write_iceberg.py

```diff
@@ -35,12 +36,20 @@ def _create_iceberg_table(
     kms_key: Optional[str] = None,
     boto3_session: Optional[boto3.Session] = None,
     dtype: Optional[Dict[str, str]] = None,
+    columns_comments: Optional[Dict[str, Any]] = None,
 ) -> None:
     if not path:
         raise exceptions.InvalidArgumentValue("Must specify table location to create the table.")
 
     columns_types, _ = catalog.extract_athena_types(df=df, index=index, dtype=dtype)
-    cols_str: str = ", ".join([f"{k} {v}" for k, v in columns_types.items()])
+    cols_str: str = ", ".join(
+        [
+            f"{k} {v}"
+            if (columns_comments is None or columns_comments[k] is None)
```

Comment:
`column_comments[k]` will fail if `column_comments` is a dict but just doesn't have an entry for `k`. You can fix this by changing this line into:
```python
if (columns_comments is None or columns_comment.get(k) is None)
```